hugr-model/src/v0/mod.rs

```diff
@@ -0,0 +1,368 @@
+//! Version 0 (unstable).
+//!
+//! # Terms
+//!
+//! Terms form a meta language that is used to describe types, parameters and metadata that
```

Comment:
This is a design point that has probably been addressed elsewhere, but do we really want structured metadata like this?

Ok, I can see many use cases for metadata which does contain....Hugr-defined things. But not necessarily Terms - maybe Hugrs or NodeId's or other things. And, how is *unstructured* metadata represented? (I see that a MetaItem contains exactly one Term, I haven't seen the alternative - am I looking in the wrong place?)

Reply:
If you want unstructured metadata, you can still put it into a string. Structured metadata allows validation via typechecking.

Reply:
Ah OK yes `String` is good, right